src/API/Site/Controllers/MerchantCenter/PhoneNumberController.php

```diff
@@ -0,0 +1,199 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Merchant;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantVerification;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Exception;
+use WP_Error;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class PhoneNumberController
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class PhoneNumberController extends BaseOptionsController {
+
+	/**
+	 * @var Merchant $merchant_verification
+	 */
+	protected $merchant_verification;
+
+	/**
+	 * PhoneNumberController constructor.
+	 *
+	 * @param RESTServer           $server
+	 * @param MerchantVerification $merchant_verification
+	 */
+	public function __construct( RESTServer $server, MerchantVerification $merchant_verification ) {
+		parent::__construct( $server );
+		$this->merchant_verification = $merchant_verification;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'mc/phone-number',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_phone_number_endpoint_read_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_phone_number_endpoint_edit_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => $this->get_update_args(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get a callback for the phone number endpoint.
+	 *
+	 * @return callable
+	 */
+	protected function get_phone_number_endpoint_read_callback(): callable {
+		return function( Request $request ) {
+			try {
+				return $this->get_phone_number_response(
+					$this->merchant_verification->get_phone_number(),
+					$request
+				);
+			} catch ( Exception $e ) {
```

Comment:
Here we only expect a `ContentApiException` to be thrown (right?) so maybe this could be modified to catch that exception instead of a generic one (+ `get_phone_number_endpoint_edit_callback`).